src/features/accounts/components/social-login/social-login.tsx

```diff
@@ -0,0 +1,292 @@
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  Box,
+  Button,
+  Center,
+  FieldWrapper,
+  GithubIcon,
+  GoogleIcon,
+  HStack,
+  IconButton,
+  Input,
+  Spinner,
+  Text,
+  TwitterIcon,
+  useToast,
+} from "components"
+import { doesAccountExist, useAccountsStore } from "features/accounts"
+import React from "react"
+import { get, useForm } from "react-hook-form"
+import { useQuery, useMutation } from "react-query"
+
+import { Ed25519KeyPairIdentity } from "@liftedinit/many-js"
+import { CHAIN_NAMESPACES, WALLET_ADAPTERS, UserInfo } from "@web3auth/base"
+import { Web3AuthCore } from "@web3auth/core"
+import {
+  OpenloginAdapter,
+  OpenloginLoginParams,
+} from "@web3auth/openlogin-adapter"
+
+const WEB3AUTH_CLIENT_ID = process.env.REACT_APP_WEB3AUTH_CLIENTID
+const WEB3AUTH_NETWORK = (process.env.REACT_APP_WEB3AUTH_NETWORK ??
+  "testnet") as "testnet" | "cyan"
+
+export const LOGIN_PROVIDER = {
+  GOOGLE: "google",
+  GITHUB: "github",
+  TWITTER: "twitter",
+  EMAIL_PASSWORDLESS: "email_passwordless",
+} as const
+
+const socials = [
+  { Icon: GoogleIcon, provider: LOGIN_PROVIDER.GOOGLE },
+  { Icon: TwitterIcon, provider: LOGIN_PROVIDER.TWITTER },
+  { Icon: GithubIcon, provider: LOGIN_PROVIDER.GITHUB },
+]
+
+export function SocialLogin({ onSuccess }: { onSuccess: () => void }) {
+  const toast = useToast()
+  const { createAccount, accounts } = useAccountsStore(
+    ({ byId, createAccount }) => ({
+      createAccount,
+      accounts: Array.from(byId),
+    }),
+  )
+  const { query, mutation } = useWeb3authProvider()
+
+  const { isFetching: isWeb3authLoading, error: web3authError } = query
+  const { mutateAsync: doConnectTo, isLoading: isConnecting } = mutation
+
+  const form = useForm({
+    defaultValues: { email: "" },
+  })
+  const { formState, register } = form
+  const { errors } = formState
+
+  async function doLogin(loginProvider: string) {
+    const isEmailPasswordless =
+      loginProvider === LOGIN_PROVIDER.EMAIL_PASSWORDLESS
+    try {
+      const { privateKey, userInfo } = await doConnectTo({
+        loginProvider,
+        loginHint: form.getValues("email"),
+        isEmailPasswordless,
+      })
+
+      const identity = Ed25519KeyPairIdentity.fromHex(privateKey.slice(0, 32))
+      const address = (await identity.getAddress()).toString()
```

Comment:
I tried to trace `identity.getAddress()` through the module stack and don't see why it's async. It looks to me like Ed25519, WebAuthn, and Anonymous are all synchronous.

Reply:
this is for `Ed25519KeyPairIdentity`
```
  async getAddress(): Promise<Address> {
    const coseKey = this.getCoseKey()
    return coseKey.toAddress()
  }
```

Reply:
Yeah, but `coseKey.toAddress` just returns an Address.

https://github.com/liftedinit/many-js/blob/cabe23955018511cb04b82932484182f3355557e/src/message/cose.ts#L168

Reply:
Oh i see what you mean now.  yeah, we could decide to make it sync.  that should be another issue for many-js